src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -246,14 +246,19 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_alias_node(
     const AliasNode& alias_node, const std::shared_ptr<TableStatistics>& input_table_statistics) {
   // For AliasNodes, just reorder/remove AttributeStatistics from the input
 
-  auto column_statistics =
-      std::vector<std::shared_ptr<BaseAttributeStatistics>>{alias_node.output_expressions().size()};
-
-  for (size_t expression_idx{0}; expression_idx < alias_node.output_expressions().size(); ++expression_idx) {
-    const auto& expression = *alias_node.output_expressions()[expression_idx];
-    const auto input_column_id = alias_node.left_input()->get_column_id(expression);
-    column_statistics[expression_idx] = input_table_statistics->column_statistics[input_column_id];
-  }
+  const auto& output_expressions = alias_node.output_expressions();
+  const auto output_expression_count = output_expressions.size();
+  auto column_statistics = std::vector<std::shared_ptr<BaseAttributeStatistics>>{output_expression_count};
+
+  alias_node.left_input()->iterate_output_expressions([&](const auto input_column_id, const auto& input_expression) {
+    for (size_t expression_idx{0}; expression_idx < output_expression_count; ++expression_idx) {
+      const auto& expression = output_expressions[expression_idx];
+      if (*expression == *input_expression) {
+        column_statistics[expression_idx] = input_table_statistics->column_statistics[input_column_id];
```

Comment:
Early `break`/`return` possible?

Reply:
Early returns do not work for multiple aliases of the same column, as in [L162](https://github.com/hyrise/hyrise/blob/cf6d05a51d968dde67cf34a808327539809e9181/resources/test_data/sqlite_testrunner_queries.sql#L162) of the SQLite tests (see below), and we get a segfault.

```SQL
SELECT a AS a1, a AS a2 FROM mixed
```